packages/compat/src/compat-app.ts

```diff
@@ -377,6 +373,13 @@ class CompatAppAdapter implements AppAdapter<TreeNames> {
     };
   }
 
+  @Memoize()
+  resolveTemplateCompilerPath() {
+    let emberSource = this.oldPackage.app.project.findAddonByName('ember-source');
+    let templateCompilerPath = emberSource.absolutePaths.templateCompiler;
+    return templateCompilerPath;
```

Comment:
This changes away from resolving the template compiler relative to the vendor asset output to resolving it in normal node modules resolution system. 

This means that it would be no longer possible for folks to provide a custom / overridden ember-template-compiler asset (before all they needed to do was put it in the right location in their vendor folder).

I think this might be perfectly fine, but I'm not 100% sure...